src/middlewared/middlewared/restful.py

```diff
@@ -532,7 +541,9 @@ def get_post_args(self, req, resp, **kwargs):
         if method.get('item_method') is True:
             method_args.insert(0, kwargs['id'])
 
-        method_kwargs = {}
+        method_kwargs = {
+            'app': Application(req.headers.get('X-Real-Remote-Addr'), req.headers.get('X-Real-Remote-Port'))
+        } if method.get('pass_application') else {}
```

Comment:
Can we please use regular syntax for things like this? Condition check happens first and then conditioned code is executed, and most people are used to read it this way.